src/lib/hash-parser-hoc.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+/* Higher Order Component to get the project id from location.hash
+ * @param {React.Component} WrappedComponent component to receive projectData prop
+ * @returns {React.Component} component with project loading behavior
+ */
+const HashParserHOC = function (WrappedComponent) {
+    class HashParserComponent extends React.Component {
+        constructor (props) {
+            super(props);
+            this.fetchProjectId = this.fetchProjectId.bind(this);
+            this.updateProject = this.updateProject.bind(this);
```

Comment:
We usually use [`lodash.bindall`](https://github.com/LLK/scratch-gui/blob/342ac4a2448f0ec6bbf5468cdd942c63d735dc34/src/containers/tag-button.jsx#L10) for this.